Minesweeper/Minesweeper/Entities/MineField.swift

```diff
@@ -0,0 +1,169 @@
+     **/
+    init(cells: [[Cell]]) {
+        self.cells = cells
+    }
+    
+    
+    /**
+     Initialise the Minefield of given square size and randomly placing the given amount of bombs.
+     - parameter size: The length of each side of the field measured in amount of cells. (Total number of cells is amount squared).
+     - parameter bombs: The amount of bombs to randomly place around in the Minefield.
+     **/
+    init(squareSideLength size: Int, withBombAmount bombs: Int) {
+        
+        let randomPlacements = MineField.getRandomMinesPlacements(mineAmount: bombs, forMinefieldTotalCellAmount: size * size)
+        var cells = [[Cell]]()
+        
+        for x in 0..<size {
+            var yCells = [Cell]()
+            
+            for y in 0..<size {
+                yCells.append(Cell(containsBomb: randomPlacements.contains(where: { placement -> Bool in
+                    return placement == (x: x, y: y)
+                })))
+            }
+            cells.append(yCells)
+        }
+        
+        MineField.markPlacedBombs(inField: &cells, placedBombs: randomPlacements)
+        
+        self.cells = cells
+    }
+    
+    
+    /**
+     Retrieve a cell's immediate neighbours (up, down, left, right & diagonally).
+     - parameter cellCoords: The X & Y position of the cell to retrieve neighbours from.
+     - parameter field: The Minefield to look through.
+     - parameter returns: All the coordinates of the given cell's neighbours (up to 8).
+     **/
+    private static func getImmediateNeighbours(ofCellAt cellCoords: Coordinates, inField field: [[Cell]]) -> [Coordinates] {
+        var neighbouringCellsCoords = [Coordinates]()
+        
+        for xDelta in -1...1 {
+            if let _ = field[safe: xDelta + cellCoords.x] {
+                let x = xDelta + cellCoords.x
+                
+                for yDelta in -1...1 {
+                    if let cell = field[x][safe: yDelta + cellCoords.y], !cell.containsBomb {
+                        neighbouringCellsCoords.append((x: x, y: yDelta + cellCoords.y))
+                    }
+                }
+            }
+        }
+        
+        return neighbouringCellsCoords
+    }
+    
+    
+    /**
+     For any given bomb, increment all neighbour's adjacent bombs count. Only increments cells that do not already contain a bomb.
+     - parameter field: The Minefield's cells to mark.
+     - parameter placedBombs: The coordinates of all the bombs to mark.
+     **/
+    private static func markPlacedBombs(inField field: inout [[Cell]], placedBombs: [Coordinates]) {
+        for placedBomb in placedBombs {
+            let immediateNeighboursCoords = MineField.getImmediateNeighbours(ofCellAt: placedBomb, inField: field)
+            for neighbourCoords in immediateNeighboursCoords {
+                
+                // we can force unwrap here, cells without a bomb
+                field[neighbourCoords.x][neighbourCoords.y].neighbouringBombs! += 1
```

Comment:
Would this not always increment the count so we always have `neighbouringBombs = neigbouringCoordinates.count`?